ros_ws/src/get_user_request.py

```diff
@@ -0,0 +1,141 @@
+    #break out and pass to publish
+    judgement=Judge(request).lower()
+    if judgement.lower()=="yes":
+        #pubish response
+        target=extract_request(request)
+
+        print("The target is ",target)
+        unknown_request=False
+    while unknown_request:
+        #add user request or claification to the conversation
+        context.append(HumanMessage(content=user_msg))
+        response=chat(context)
+        print( response.content)
+        attempts_to_understand+=1
+
+        #seek input from user 
+        request=input("User:")
+        #If response sufficiently refined 
+        #break out and pass to publish
+        judgement=Judge(request).lower()
+        if request.lower()=="yes":
+            target=extract_request(response.content)
+            print("The target is ",target)
+            break
+        if judgement=="yes":
+            #identify simple target
+            target=extract_request(request)
+            #pubish target
+            print("The target is ",target)
+            break
+        #if there have been too many cycles restart the request
+        if attempts_to_understand==5:
+            print("I'm sorry i dont inderstand.  Could we start over?")
+            break
+        context.append(response)
+
+def Judge(request):
+    #llm = OpenAI(model_name=model_to_use, temperature=0)
+    llm = ChatOpenAI(model_name=model_to_use,temperature=0)
+
+    context="""An agent is attempting to get a specific item requested by a human user.  
+        It should be an object found in a house or apartment.
+        The human user is sometimes vague and the agent is trying to clarify specifically what the 
+        human user wants.  You are an AI helping the agent to judge if the item 
+        requested is specific enough to search for.  
+        You are also an expert in grammatical structures and will be given a sentence from which 
+        you should extract what exactly is being requested. It will likely be the direct object in a request 
+        or question or the subject of a statement.  
+        If the response has the word "anything" or "something" in it, it is not specific enough.
+        If the object is specific enough to search for, respond  with "yes", otherwise respond with "no".  Do not be verbose. 
+        """
+    
+    system_message_prompt=SystemMessagePromptTemplate.from_template(context)
+    human_template=f"Am I specifically requesting an object, if so what? : {request}"
+    human_message_prompt=HumanMessagePromptTemplate.from_template(human_template)
+
+    chat_prompt=ChatPromptTemplate.from_messages([system_message_prompt,human_message_prompt])
+    chain=LLMChain(llm=llm, prompt=chat_prompt)
+    response=chain.run(question=request)
+    response=cleanup(response)
+    return response
+
+def extract_request(verbose_request):
+    #llm = OpenAI(model_name=model_to_use, temperature=0)
+    llm = ChatOpenAI(model_name=model_to_use,temperature=0)
```

Comment:
We're recreating the chat API each time, when, if this was an object, we could instantiate the entire entity as a singular object with configuration set at instantiation, and then can skip much of this within the functions, making it far simpler.

Reply:
I had these separate bc they needed slightly different context personalities.  But if I created an LLM object, then had the different contexts in files to feed, When I called created an instance in the refiner, for example, I instantiate with the proper context file and then I instantiate another instance for the judge with its context file- is that what you're suggesting?

Reply:
Does the `ChatOpenAI` object maintain conversational history and thus context? I thought it did not.